cg/meta/archive/archive.py

```diff
@@ -20,24 +39,24 @@ def __init__(self, housekeeper_api: HousekeeperAPI, status_db: Store):
         self.housekeeper_api: HousekeeperAPI = housekeeper_api
         self.status_db: Store = status_db
 
-    def get_files_by_archive_location(
-        self, file_data: List[PathAndSample], archive_location: ArchiveLocationsInUse
-    ) -> List[PathAndSample]:
-        """
-        Returns a list of PathAndSample where the associated sample has a specific archive location.
-        """
-        selected_files: List[PathAndSample] = []
-        for file in file_data:
-            sample: Optional[Sample] = self.get_sample(file)
-            if sample and sample.archive_location == archive_location:
-                selected_files.append(file)
-        return selected_files
-
-    def get_sample(self, file: PathAndSample) -> Optional[Sample]:
-        sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(file.sample_internal_id)
+    def get_sample(self, file: File) -> Optional[Sample]:
+        """Fetches the Sample corresponding to a File and logs if a Sample is not found."""
+        sample: Optional[Sample] = self.status_db.get_sample_by_internal_id(
+            file.version.bundle.name
+        )
         if not sample:
             LOG.warning(
-                f"No sample found in status_db corresponding to sample_id {file.sample_internal_id}."
+                f"No sample found in status_db corresponding to sample_id {file.version.bundle.name}."
                 f"Skipping archiving for corresponding file {file.path}."
             )
         return sample
+
+    def add_samples_to_files(self, files_to_archive: List[File]) -> List[FileAndSample]:
+        """Fetches the Sample corresponding to each File, instantiates a FileAndSample object and
+        adds it to the list which is returned."""
+        files_and_samples: List[FileAndSample] = []
+        for file in files_to_archive:
+            sample: Optional[Sample] = self.get_sample(file)
+            if sample:
```

Comment:
Could this filtering be done earlier in the flow? Maybe it would make the rest of the flow easier - just a thought